controller/konnect/ops/ops_errors.go

```diff
@@ -123,3 +127,42 @@ func SDKErrorIsConflict(sdkError *sdkkonnecterrs.SDKError) bool {
 	}
 	return false
 }
+
+// handleDeleteError handles errors that occur during a delete operation.
+// It logs a message and returns nil if the entity was not found in Konnect (when
+// the delete operation is skipped).
+func handleDeleteError[
+	T constraints.SupportedKonnectEntityType,
+	TEnt constraints.EntityType[T],
+](ctx context.Context, err error, ent TEnt) error {
+	logDeleteSkipped := func() {
+		ctrllog.FromContext(ctx).
+			Info("entity not found in Konnect, skipping delete",
+				"op", DeleteOp,
+				"type", ent.GetTypeName(),
+				"id", ent.GetKonnectStatus().GetKonnectID(),
+			)
+	}
+
+	var sdkNotFoundError *sdkkonnecterrs.NotFoundError
+	if errors.As(err, &sdkNotFoundError) {
+		logDeleteSkipped()
+		return nil
+	}
+
+	var sdkError *sdkkonnecterrs.SDKError
+	if errors.As(err, &sdkError) {
+		if sdkError.StatusCode == 404 {
```

Comment:
```suggestion
		if sdkError.StatusCode == http.StatusNotFound {
```
It is interesting that linters miss this

Reply:
There is such a linter that checks for "magic numbers". But in its default settings, only `0` is not treated as a magic number: https://github.com/tommy-muehle/go-mnd

Reply:
It can be enabled with `golangci-lint`: https://golangci-lint.run/usage/linters/#mnd 👍

Reply:
Maybe it is worth enabling it and see how it behaves